src/solvers/flattening/arrays.cpp

```diff
@@ -564,17 +524,9 @@ void arrayst::add_array_constraints_with(
 
       if(guard_lit!=const_literal(true))
```

Comment:
Just curious: would it also make sense to special-case `guard_lit==const_literal(false)`?